File: client/src/types/Product.ts
export interface Product {
  id: number;
  name?: string | null;
  sku?: string | null;
  product_type?: string | null; // corresponds to DB column "type"
  category?: string | null;
  price: number;
  description?: string | null;
  partner_discount?: number | null;
}

File: api/schemas/product.py
from typing import Optional
from pydantic import BaseModel


class ProductSchema(BaseModel):
    id: int
    name: Optional[str] = None
    sku: Optional[str] = None
    product_type: Optional[str] = None  # maps to DB column "type"
    category: Optional[str] = None
    price: float
    description: Optional[str] = None
    partner_discount: Optional[float] = None

    class Config:
        from_attributes = True
